indico/web/client/js/custom_elements/ind_datepicker.js

```diff
@@ -0,0 +1,557 @@
+        const markEnd = rangeEnd || (rangeStart && selectedDate);
+
+        for (let i = 0; i < CALENDAR_CELL_COUNT; i++) {
+          const date = new Date(firstDayOfCalendar);
+          date.setDate(date.getDate() + i);
+          const $calendarButton = listbox.querySelector(`button:nth-of-type(${i + 1})`);
+          $calendarButton.textContent = date.getDate();
+          $calendarButton.dataset.currentMonth = date.getMonth() === month;
+          $calendarButton.setAttribute(
+            'aria-label',
+            date.toLocaleDateString(indCalendar.locale, {
+              year: 'numeric',
+              month: 'long',
+              day: 'numeric',
+            })
+          );
+          $calendarButton.toggleAttribute(
+            'data-weekend',
+            indCalendar.weekInfo.weekend.includes(date.getDay() + 1)
+          );
+          $calendarButton.tabIndex = -1; // reset tabIndex
+          $calendarButton.value = toDateString(date);
+
+          // Mark range
+          $calendarButton.disabled = date < rangeStart || date > rangeEnd;
+          $calendarButton.toggleAttribute('data-range-start', sameDate(date, markStart));
+          $calendarButton.toggleAttribute('data-range-end', sameDate(date, markEnd));
+          $calendarButton.toggleAttribute('data-range', date < markEnd && date > markStart);
+
+          // Mark selected
+          if ($calendarButton.value === indCalendar.value) {
+            $calendarButton.setAttribute('aria-selected', true);
+          } else {
+            $calendarButton.removeAttribute('aria-selected');
+          }
+        }
+
+        const $focusableButton =
+          listbox.querySelector('[aria-selected]') || listbox.firstElementChild;
+        $focusableButton.tabIndex = 0;
+      }
+    }
+
+    disconnectedCallback() {
+      this.cleanup();
+    }
+
+    static observedAttributes = ['open', 'value', 'range-start', 'range-end'];
+
+    attributeChangedCallback(name) {
+      switch (name) {
+        case 'open':
+          this.dispatchEvent(new Event('attrchange.open'));
+          break;
+        case 'value':
+        case 'range-start':
+        case 'range-end':
+          this.dispatchEvent(new Event('attrchange.value'));
+          break;
+      }
+    }
+  }
+);
+
+// React hacks
+
+function setNativeInputValue(input, value) {
+  // React adds its own setter to the input and messes with the native event mechanism.
+  // In order for the value to be set in a standard way, we need to resort to this hack.
+  Object.getOwnPropertyDescriptor(HTMLInputElement.prototype, 'value').set.call(input, value);
+}
+
+// Date functions
+
+function getToday() {
+  const now = new Date();
+  now.setHours(0, 0, 0, 0);
+  return now;
+}
+
+function sameDate(a, b) {
+  return a?.getTime() === b?.getTime();
+}
+
+function toDateString(date) {
+  // We do not use ISO date format because it will be interpreted as UTC
+  return date?.toLocaleDateString('en-US');
```

Comment:
> Any way we can avoid putting a non-ISO format in the value of this field, and work around the "will be interpreted as UTC" problem?

These values are used internally. Since the `Date()` constructor parses that format consistently irrespective of the browser's language settings, and it doesn't suffer form the UTC issue, I don't think we should complicate things just for aesthetics.

> It's usually safe to assume that what you select is in the event's timezone since that typically makes sense, but this is not something we can (from a technical point of view) assume / rely on.

The datepicker code is should be time zone agnostic, if that's your concern?

Reply:
That's fine then (I was indeed hoping for it to be TZ agnostic).

So the `.value` of the `<ind-datepicker>` should be considered internal. I was looking at it like a form input field, but maybe I shouldn't do that (or maybe a different name may be better, since I could imagine someone else making the same mistake). If it's internal then I'd say it's fine as long as we do not end up with cases where `mm/dd/yyyy` vs `dd/mm/yyyy` can cause problems

Reply:
Alternatively we can probably use the RFC date format which is obtained by calling `Date.prototype.toDateString()`, tho.